### PR TITLE
Added methods for getting user's subscribed, contributed to, and moderated, subreddits

### PR DESCRIPTION
**Pull request by mleef**
Implements: 
- [GET subreddits/mine/contributor](http://www.reddit.com/dev/api#GET_subreddits_mine_contributor)
- [GET subreddits/mine/subscriber](http://www.reddit.com/dev/api#GET_subreddits_mine_subscriber)
- [GET subreddits/mine/moderator](http://www.reddit.com/dev/api#GET_subreddits_mine_moderator)

### CODE REVIEW

**Comment by sfat** [comment]
@mleef can we try adding some unit tests for this puppy. Anyway, thanks for the contribution !

**Comment by mleef** [comment]
@sfat Will do, thanks for the feedback.

File: src/main/java/com/github/jreddit/entity/User.java [hunk 1]
```diff
@@ -107,4 +111,56 @@ public void connect() throws IOException, ParseException {
...
+     */
+    public List<Subreddit> getModeratorOf() throws RetrievalFailedException, RedditError {
+        if (this.getCookie() == null || this.getModhash() == null) {
+            System.err.printf("Please invoke the connect method in order to login the user");
+            return null;
+        }
+        Subreddits sub = new Subreddits(restClient, this);
+
+        // Automatically returns maximum allowed number of objects, should this be a parameter?
+        return sub.parse(ApiEndpointUtils.USER_GET_MODERATOR_OF + "?&limit=100");
```
Comment (sfat) [inline src/main/java/com/github/jreddit/entity/User.java]: I am not sure about this. I'm thinking maybe leave the limit=100 as default, and have the possibility to add limit if needed. i.e. if limit == null { limit = 100 }

File: src/main/java/com/github/jreddit/entity/User.java [hunk 2]
```diff
@@ -107,4 +111,56 @@ public void connect() throws IOException, ParseException {
...
+     */
+    public List<Subreddit> getSubscribed() throws RetrievalFailedException, RedditError {
+        if (this.getCookie() == null || this.getModhash() == null) {
+            System.err.printf("Please invoke the connect method in order to login the user");
+            return null;
+        }
+        Subreddits sub = new Subreddits(restClient, this);
+
+        // Automatically returns maximum allowed number of objects, should this be a parameter?
+        return sub.parse(ApiEndpointUtils.USER_GET_SUBSCRIBED + "?&limit=100");
```
Comment (sfat) [inline src/main/java/com/github/jreddit/entity/User.java]: see below comment

File: src/main/java/com/github/jreddit/entity/User.java [hunk 3]
```diff
@@ -107,4 +111,56 @@ public void connect() throws IOException, ParseException {
...
+     */
+    public List<Subreddit> getContributedTo() throws RetrievalFailedException, RedditError {
+        if (this.getCookie() == null || this.getModhash() == null) {
+            System.err.printf("Please invoke the connect method in order to login the user");
+            return null;
+        }
+        Subreddits sub = new Subreddits(restClient, this);
+
+        // Automatically returns maximum allowed number of objects, should this be a parameter?
+        return sub.parse(ApiEndpointUtils.USER_GET_CONTRIBUTED_TO + "?&limit=100");
```
Comment (sfat) [inline src/main/java/com/github/jreddit/entity/User.java]: see below comment

**Comment by mleef** [comment]
Updated methods by adding limit parameter and unit tests for each, not sure why the build is failing, all of the tests are working locally...

**Comment by sfat** [comment]
@mleef think it was some hiccup from travis. I ran the build again and it seems to be okay. Merging. thanks!

**Pull request merged**